### ISSUE TITLE
Communication between pods should happen through SSL on Openshift/K8S when I set "secured"service annotation to true

### ISSUE THREAD

**Issue by fgapito** [spring-cloud/spring-cloud-kubernetes#1141]
Hi,

I'm facing with this possible issue when I try to establish a https/ssl connection from A->B (B has secured annotation on k8s service annotation). It seems that the annotation it is loaded but it is not used to set secure = true inside the KubernetesServiceInstance object. I checked in the KubernetesInformerDiscoveryClient and I see this at line 163 (v 2.1.3)

![image](https://user-images.githubusercontent.com/70958010/202775250-af428edc-3f89-4775-addb-9fbf1aebcf4c.png)

You can see that secure constructor parameter is always **false**.

Is it normal? Why, even if I put annotation on service description, the service B is called through http instead of https?

This is the reference to the documentation where secured annotation is mentioned:
![image](https://user-images.githubusercontent.com/70958010/202775961-276f5213-ef45-4d87-86ce-0af751ee1458.png)

https://docs.spring.io/spring-cloud-kubernetes/docs/2.1.1/reference/html/#loadbalancer-for-kubernetes

Thank you

Kind Regards

f

**Comment by ryanjbaxter** [comment]
That doc is specific to the load balancer not service discovery.

Do you have an https port defined in your service resource?
See the bottom of this section of the docs https://docs.spring.io/spring-cloud-kubernetes/docs/2.1.1/reference/html/#discoveryclient-for-kubernetes

**Comment by fgapito** [comment]
Yes I have it and the name is "main". I also configured spring cloud discovery to use main as primary port with the appropriate configuration. I also tried renaming that port in "Https". I hacked that class reading the secured annotation value and it works but this is just a bad patch and I do not want to put that in production.

**Comment by ryanjbaxter** [comment]
Can you tell me how you are using the discovery client to make the request?

**Comment by fgapito** [comment]
I'm using these deps:

```
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-kubernetes-client-discovery</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-loadbalancer</artifactId>
        </dependency>
```

The rest call is executed using Feign.

The client spring boot service has this configuration:
```

spring:
  cloud:
    kubernetes:
      discovery:
        primary-port-name: 'main'
      config:
        enabled: false

```

this is the serice definition of the target service to be called with https

```
apiVersion: v1
kind: Service
metadata:
  name: target-svc
  annotations:
    secured: 'true'
spec:
  ports:
    - name: main
      protocol: TCP
      port: 44389
      targetPort: 44389
  selector:
    app: target-svc

```

Let me know if I can give you more info.

F

**Comment by ryanjbaxter** [comment]
Thanks.  How are you making the http request to the service?

**Comment by fgapito** [comment]
Using feign

**Comment by fgapito** [comment]
This is the change that let the https call happen:

![image](https://user-images.githubusercontent.com/70958010/204013504-0ba5cc56-9bcb-455f-8337-51713b5a3822.png)

**Comment by ryanjbaxter** [comment]
Would you be interested in submitting a PR?

**Comment by fgapito** [comment]
Of course, I'm going to do that.

Thank you.

**Comment by fgapito** [comment]
Here the PR requested.

https://github.com/spring-cloud/spring-cloud-kubernetes/pull/1150

Thank you.

**Comment by fgapito** [comment]
HI,

do you know approximatively when the spring-cloud-kubernetes version 2.1.6 (that should contains this fix) will be pushed on official maven repositories?

thank you.

f

**Comment by ryanjbaxter** [comment]
@fgapito you can find tentative release dates here https://github.com/spring-cloud/spring-cloud-release/milestones

**Comment by fgapito** [comment]
Hi, I tought In 2021.0.5 but it seems not. Did something go wrong or should I wait for 2021.0.6?
f

**Comment by ryanjbaxter** [comment]
2021.0.5 was released on November 3rd, we didn't merge this till December 6th it will be included in 2021.0.6
